qiskit_ibm_runtime/qiskit_runtime_service.py

```diff
@@ -1316,29 +1319,19 @@ def delete_job(self, job_id: str) -> None:
             job_id: ID of the job to delete.
 
         Raises:
-            RuntimeJobNotFound: If the job doesn't exist.
-            IBMRuntimeError: If the request failed.
+            IBMInputValueError: Method is not supported.
         """
-        try:
-            self._active_api_client.job_delete(job_id)
-        except RequestsApiError as ex:
-            if ex.status_code == 404:
-                raise RuntimeJobNotFound(f"Job not found: {ex.message}") from None
-            raise IBMRuntimeError(f"Failed to delete job: {ex}") from None
+        raise IBMInputValueError(
+            "This method is not supported on the new IBM Quantum Platform. "
+            "It will be deprecated and removed in a future release."
+        )
```

Comment:
I'm not sure how to deal with the `delete_job()` method. It's not supported on the server side so should we deprecate it first? Or just raise an error for now.

Reply:
I think that the deprecation warning + error combination works for now.